src/compiler/types.ts

```diff
@@ -572,6 +574,11 @@ module ts {
         typeName: EntityName;
         typeArguments?: NodeArray<TypeNode>;
     }
+    
+    export interface TypeGuardTypeNode extends TypeNode {
+        target?: Identifier;

```

Comment:
Yeah I'd rather have target be of type `Identifier | ThisKeyword`. Oh, but ThisKeyword is not a node... I don't have a better idea